### PR TITLE
Implement client side for Go panel

### PR DESCRIPTION
**Pull request by tttzach**
Implement UI for Go panel.

Screenshot 2020-08-04 at 11 39 12 PM

### CODE REVIEW

File: src/main/webapp/index.html [hunk 1]
```diff
@@ -166,13 +166,35 @@ <h2 class="panel-row__header u-align-with-first-panel">Build your plan:</h2>
     <div class="panel-row__content">
       <div class="panel">Assign</div>
       <div class="panel">Plan</div>
-      <div class="panel">Go
-        <div id="go"></div>
+      <div class="panel">
```
Comment (austinteshuba) [inline src/main/webapp/index.html:169]: Can you create a utility class `.u-border-go` and change the border of this panel to match some unique color (I'm thinking a material green for this panel - currently saved as a variable actually `--color-green`). Any color is good, but I think the black looks too plain for our colour palette.

File: src/main/webapp/style/base/_typography.css [hunk 2]
```diff
@@ -49,6 +49,13 @@ body {
   width: var(--panel-icon-width);
 }
 
+.panel__optimize-button {
```
Comment (austinteshuba) [inline src/main/webapp/style/base/_typography.css:52]: So this is coming to master soon in PR #125, but there are some existing buttons you can put at the bottom of this panel. 

Alternatively, you can use this button no problem, but I would try to fix up the alignment. It seems to not be vertically centered in its container

Comment (austinteshuba) [reply src/main/webapp/style/base/_typography.css:52]: Also, it may be better to move this button to the bottom of the panel. If a tasklist name is long, it might push your button off the panel or otherwise squash it. 

You could even use the "Push to Start" button that's currently in the "Assign" panel and just create a content entry for this.

File: src/main/webapp/index.html (hunk 1 above)
Comment (tttzach) [reply src/main/webapp/index.html]: Got it - changed to orange. Thanks!

File: src/main/webapp/style/base/_typography.css (hunk 2 above)
Comment (tttzach) [reply src/main/webapp/style/base/_typography.css:54]: I think this is slightly better now(?)

Screenshot 2020-08-06 at 2 18 36 AM

Comment (austinteshuba) [reply src/main/webapp/style/base/_typography.css]: Yep it is! Small nit picks: 
1) Make the button green (there's an existing class for this, `panel__green-button`)
2) If you want, if you found a way to overwrite the `width` property of the button to be `100%` (you could make another class for this if you wanted by adopting most of the `button-1-of-2` properties and changing the width. The new class could be `button-1-of-1`)

Comment (gbuenoandrade) [reply src/main/webapp/style/base/_typography.css]: +1 to making the optimize button green

**Comment by tttzach** [comment]
> LGTM
> 
> Though I think it would be cool if you could give the user some sort of feedback after they press "optimize." Maybe just write "..." in the meantime.

Good idea!

Before pressing optimize
Screenshot 2020-08-06 at 5 36 42 PM

After pressing optimize
Screenshot 2020-08-06 at 5 39 31 PM

File: src/main/webapp/style/base/_typography.css (hunk 2 above)
Comment (tttzach) [reply src/main/webapp/style/base/_typography.css]: Thanks for the suggestions! I made a new class and put 30% instead - not a designer by any means but I just thought 100% made it look a little weird.

Screenshot 2020-08-06 at 5 38 52 PM

File: src/main/java/com/google/sps/servlets/GoServlet.java [hunk 3]
```diff
@@ -148,7 +148,7 @@ public void doGet(
     List<Task> tasks;
 
     String taskLists = request.getParameter("taskLists");
-    if (taskLists == null) {
+    if (taskLists.isEmpty()) {
```
Comment (antmar) [inline src/main/java/com/google/sps/servlets/GoServlet.java:151]: I am not sure what prompted this change, but you may need to handle both null and empty here?

For future reference, the Apache Commons package has a utility method called `StringUtils.isEmpty` that handles `null` the same as empty.

Comment (tttzach) [reply src/main/java/com/google/sps/servlets/GoServlet.java:151]: I found that because `taskLists=` will always be in the query string, getting the `taskLists` parameter will return `""` instead of `null`. However, I changed it to `StringUtils.isEmpty` anyway just so I'll remember it exists next time. Thanks!

File: src/main/webapp/auth.js [hunk 4]
```diff
@@ -79,13 +79,10 @@ function handleAuthenticationState() {
     // Populate information panels at top of dashboard
     populateGmail();
     populateTasks();
-    populateCalendar();
+    // populateCalendar();
```
Comment (antmar) [inline src/main/webapp/auth.js:82]: Um, did you intend to comment this out? It seems unrelated to the Go panel.

Comment (tttzach) [reply src/main/webapp/auth.js:82]: Yeah sorry about that, I wanted to screen capture the deployed site and this function writes a bunch of holidays from calendar behind the Go panel. However, we just decided it's possible to merge everything and then do our screen capture off the final product.

File: src/main/java/com/google/sps/servlets/GoServlet.java (hunk 3 above)
Comment (tttzach) [reply src/main/java/com/google/sps/servlets/GoServlet.java]: `RESOLVED`

File: src/main/webapp/auth.js (hunk 4 above)
Comment (tttzach) [reply src/main/webapp/auth.js]: `RESOLVED`
